pkg/openstack/loadbalancer.go

```diff
@@ -401,15 +407,13 @@ func memberExists(members []v2pools.Member, addr string, port int) bool {
 }
 
 func popListener(existingListeners []listeners.Listener, id string) []listeners.Listener {
-	for i, existingListener := range existingListeners {
-		if existingListener.ID == id {
-			existingListeners[i] = existingListeners[len(existingListeners)-1]
-			existingListeners = existingListeners[:len(existingListeners)-1]
-			break
+	newListeners := []listeners.Listener{}
```

Comment:
old solution did not work for me. I created 5 loadbalancers which had multiple listeners. For some reason in some of them this code did not work properly. That is why I wrote tests for it and updated the solution little bit.

Reply:
so you might suffering race here with old solution?

Reply:
Yeah, the code here looks not very clear. Glad to see the readability improved.